test-resources/integration-tests/helpers/jwt-verifier.ts

```diff
@@ -21,44 +25,55 @@ export enum ClaimNames {
 
 export class JwtVerifier {
     static ClaimNames = ClaimNames;
-    constructor(private jwtVerifierConfig: JwtVerificationConfig) {}
+    constructor(
+        private jwtVerifierConfig: JwtVerificationConfig,
+        private logger: Logger,
+    ) {}
 
     public async verify(
         encodedJwt: Buffer,
         mandatoryClaims: Set<string>,
         expectedClaimValues: Map<string, string>,
     ): Promise<JWTVerifyResult | null> {
-        try {
-            const signingPublicJwkBase64 = this.jwtVerifierConfig.publicSigningJwk;
-            const signingAlgorithm = this.jwtVerifierConfig.jwtSigningAlgorithm;
-            const signingPublicJwk = JSON.parse(Buffer.from(signingPublicJwkBase64, "base64").toString("utf8"));
-            const publicKey = await importJWK(signingPublicJwk, signingPublicJwk?.alg || signingAlgorithm);
-
-            const jwtVerifyOptions = this.createJwtVerifyOptions(signingAlgorithm, expectedClaimValues);
-            const verifyResult = await jwtVerify(encodedJwt, publicKey, jwtVerifyOptions);
+        if (!mandatoryClaims || mandatoryClaims?.size === 0) throw new Error("No mandatory claims provided");
```

Comment:
I didn't put error handling in this because I expected it to be caught by teest assertions. Keen to understand why you disagree?

Reply:
No, I copied over the original jwtVerifier I played around with the `getPublicKeyUsingJwksUri` in my stack first